README.md

```diff
@@ -51,6 +58,20 @@ Here are the main steps outlined.
 
 To learn about how to write your configuration, please refer to our [`.spark.example.yml` file](https://github.com/BluesparkLabs/spark/blob/master/.spark.example.yml).
 
+**5. Create a file named `.env` in your project's root.** This will be your environment-specific configuration that Spark will use. Don't commit this file to your repository, it is meant to hold values that are not supposed to transfer between environments. Set a variable named `SPARK_MODE` inside the file. See more in the chapter about [Environment modes](#environment-modes).
+
+### Environment modes
+
+Spark can provide an environment with or without containers for PHP and the HTTP server. Our intent with that is to provide a sane—though not ideal—development environment on macOS. Those who are lucky enough to enjoy Docker on Linux can run the whole environment in containers, but developers on macOS are encouraged to set up their own PHP with a web server and leave only the rest (database, Solr etc. to Spark). (Our recommendation is to use Apache that ships with macOS and install PHP packages from https://php-osx.liip.ch.)
+
+To configure your environment mode, **create a file named `.env`** in your project's root and **set a variable named `SPARK_MODE`** to one of the following values:
+
+* `containers`
+* `containers--no-php-and-http-server`
+* `no-containers`
+
+When no containers are used, Spark will assume that PHP and the HTTP server runs locally, and your application takes care of connecting to other services, i.e. the database. `no-containers` is the fallback value when the `.env` file or the `SPARK_MODE` variable is missing.
```

Comment:
I'm a little concerned that this is going to really confuse people.  We need to elaborate on the actual setup steps required to get a *sane though not ideal development environment on macOS working.  

For example, if you use containers--no-php-and-http-server  you cant really use the settings.spark.php file, because you have to know the external ip and port assigned by docker for the db connection, and I assume put that  in settings.local.php, right?    This probably needs explaining, and ideally there needs to be a stupid simple way to generate that or something.

Reply:
`settings.spark.php` takes care of the `containers--no-php-and-http-server` use case completely.

I strongly feel that we should not spend time on providing setup steps for an environment on macOS. I already provided some recommendations in a short sentence, and there are many tutorials on the internet about this topic. It could be done as a blog post or something, but let's please not worry about that in Spark, as we need to continue making progress on more important stuff.

With that said, I can work on making the wording better.